maven_iuvs/miscellaneous.py

```diff
@@ -122,21 +131,28 @@ def orbit_folder(orbit):
     return f"orbit{orbit_set:05}"
 
 
-def iuvs_orbno_from_fname(fname):
+def iuvs_orbno_from_fname(fname, search_full_path=False):
     """
     Collects the orbit number from the filename.
     Parameters
     ----------
     fname : string
             IUVS observation filename
+    search_full_path : boolean
+                       if True, this function will use regular expressions to search the full file path.
+                       Avoids problem where searching the full path using split() returns only the orbit set 
+                       (i.e. 16900 instead of 16910)
 
     Returns
     -------
     orb_string : int
                  orbit number
     """
-    orb_string = str(fname).split('orbit')[1][:5]
-    return int(orb_string)
+    if search_full_path:
+        return int(re.search(orbno_RE, fname).group(0))
```

Comment:
seems to me that using os.path.basename would be clearer and platform-independent…

Reply:
sounds good! Made the change

Reply:
committed?

Reply:
Sorry, thought I pushed but hadn't. Fixed by [6488aa7](https://github.com/lasp/maven_iuvs/pull/63/commits/6488aa7a09b1698c3a6508eee4bfd5b68241f8c6)!